### PR TITLE
AnimeFenix

### PR DESCRIPTION
**Pull request by Keiser04**
I still have to figure out how to add this because I don't understand how to do it.
image

### CODE REVIEW

File: src/pages/AnimeFenix/main.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  isOverviewPage: (url: string) => {
+    let path = url.split('/')[3];
+    if (path && path !== 'ver' && path !== 'zerotwo') {
+      return true;
+    }
+    return false;
+  },
+  sync: {
+    getTitle: (url: string) => {
+      let titleElement = document.querySelector("body > div.hero.is-fullheightX > section > div > div > div.column.is-12-mobile.is-8-tablet.is-9-desktop > h1");
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:23]: Selector to specific. Breaks with every page change

File: src/pages/AnimeFenix/main.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      let title = titleElement?.textContent ? titleElement.textContent.replace(/\d+\s*(Sub|Dub|Doblaje|Español)$/i, '').trim() : null;
+      
+      if (!title) {
+        let urlTitle = url.split('/')[4];
+        title = urlTitle ? urlTitle.replace(/-/g, ' ').replace(/\b\w/g, l => l.toUpperCase()) : '';
+      }
+      return title.split(' (')[0];
+    },
+    getIdentifier: (url: string) => {
+      let titleElement = document.querySelector("body > div.hero.is-fullheightX > section > div > div > div.column.is-12-mobile.is-8-tablet.is-9-desktop > h1");
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:33]: Duplicate code

File: src/pages/AnimeFenix/main.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return;
+      }
+      if (page.url.split('/')[3] === 'ver' || page.url.split('/')[3] !== 'zerotwo') {
+        page.handlePage();
+      }
+    });
+  },  
+};
+
+export interface pageState {
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:70]: What does this do?

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Comment by lolamtisch** [comment]
You need to setup the overiview correctly with the uiSelector correctly setuped

**Comment by Keiser04** [comment]
> You need to setup the overiview correctly with the uiSelector correctly setuped

but is not optional?
image
image
you have a easy example?

**Comment by lolamtisch** [comment]
Yes it is optional. But if you want the ui you will need to add the overview page, but you dont need to.

sync page: the page with the player/video

**Comment by lolamtisch** [comment]
Yes it is optional. But if you want the ui you will need to add the overview, but you dont need to. If you only want to implement syncing of the watched ep you dont need to overview

sync page: video page
overview page: the page where the list of episodes is

Simple example: https://github.com/MALSync/MALSync/blob/master/src/pages/Gogoanime/main.ts

**Comment by Keiser04** [comment]
> Yes it is optional. But if you want the ui you will need to add the overview, but you dont need to. If you only want to implement syncing of the watched ep you dont need to overview
> 
> sync page: video page overview page: the page where the list of episodes is
> 
> Simple example: https://github.com/MALSync/MALSync/blob/master/src/pages/Gogoanime/main.ts

ok i will try this nigth

**Comment by Keiser04** [comment]
> Yes it is optional. But if you want the ui you will need to add the overview, but you dont need to. If you only want to implement syncing of the watched ep you dont need to overview
> 
> sync page: video page overview page: the page where the list of episodes is
> 
> Simple example: https://github.com/MALSync/MALSync/blob/master/src/pages/Gogoanime/main.ts

image
I added it but I can't get the list of episodes, I don't know what I can do to fix it, sorry.

**Comment by Keiser04** [comment]
@lolamtisch what i do? another push?

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Comment by Keiser04** [comment]
@lolamtisch I surrendered on the overview . I only have this left that I didn't know what you meant.

{
  "urls": {
    "match": ["*://animefenix.tv/*"]
  }
}

**Comment by Keiser04** [comment]
@lolamtisch can you watch if all is correct?

File: src/pages/AnimeFenix/main.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        : null;
+    
+      if (!title) {
+        const urlTitle = url.split('/')[4];
+        title = urlTitle ? urlTitle.replace(/-/g, ' ').replace(/\b\w/g, l => l.toUpperCase()) : '';
+      }
+      return title.replace(/\d+ Sub Español$/, '').trim();
+    },
+    getIdentifier: (url: string) => {
+      return AnimeFenix.sync.getTitle(url);
```
[the remark this answers is not in the document]
Comment (Keiser04) [reply src/pages/AnimeFenix/main.ts:30]: i give up whit overview

File: src/pages/AnimeFenix/main.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    },          
+    getOverviewUrl: (url: string) => {
```
[the remark this answers is not in the document]
Comment (Keiser04) [reply src/pages/AnimeFenix/main.ts:32]: i give up whit overview

File: src/pages/AnimeFenix/meta.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "urls": {
```
[the remark this answers is not in the document]
Comment (Keiser04) [reply src/pages/AnimeFenix/meta.json:2]: what do you mean @lolamtisch ?

File: src/pages/AnimeFenix/tests.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+  "title": "AnimeFenix",
+  "url": "https://animefenix.tv/",
+  "testCases": [
+    {
+      "url": "https://animefenix.tv/ver/youkoso-jitsuryoku-shijou-shugi-no-kyoushitsu-e-3rd-season-9",
```
[the remark this answers is not in the document]
Comment (Keiser04) [reply src/pages/AnimeFenix/tests.json:6]: i give up whit overview

**Review by Keiser04 — COMMENTED** [review]
i give up whit overview

File: src/pages/AnimeFenix/meta.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (lolamtisch) [reply src/pages/AnimeFenix/meta.json:2]: Like mentioned in the old PR

> Search url is missing
>
> Example: https://github.com/MALSync/MALSync/blob/master/src/pages/AnimeXin/meta.json#L2

File: src/pages/AnimeOnegai/meta.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
-{
```
Comment (lolamtisch) [inline src/pages/AnimeOnegai/meta.json:1]: Please dont commit unrelated files

File: src/pages/OtakuFR/tests.json [hunk 9]
```diff
@@ -1,30 +1,30 @@
-{
```
Comment (lolamtisch) [inline src/pages/OtakuFR/tests.json:1]: Please dont commit unrelated files

File: src/pages/AnimeFenix/main.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  isSyncPage: (url: string) => {
+    const path = url.split('/')[3];
+    return path === 'ver';
+  },
+  isOverviewPage: (url: string) => {
+    const path = url.split('/')[3];
+    return Boolean(path && path !== 'ver' && path !== 'zerotwo');
+  },
+  sync: {
+    getTitle: (url: string) => {
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:17]: title wrong

File: src/pages/AnimeFenix/main.ts [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+export const AnimeFenix: pageInterface = {
+  name: 'AnimeFenix',
+  domain: 'https://animefenix.tv',
+  languages: ['Spanish'],
+  type: 'anime',
+  isSyncPage: (url: string) => {
+    const path = url.split('/')[3];
+    return path === 'ver';
+  },
+  isOverviewPage: (url: string) => {
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:12]: Not used

File: src/pages/AnimeFenix/main.ts [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    getEpisode: (url: string) => {
+      const urlParts = url.split('/');
+      const lastPart = urlParts.pop();
+      const episodeNumber = lastPart ? lastPart.match(/\d+/g) : null;
+      return episodeNumber && episodeNumber.length > 0
+        ? parseInt(episodeNumber[episodeNumber.length - 1])
+        : 0;
+    },
+  },
+  init: (page) => {
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:48]: Style import is missing

File: src/pages/AnimeFenix/main.ts [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        h1Element.textContent &&
+        h1Element.textContent.includes('Animes Populares')
+      ) {
+        console.error('404');
+        return;
+      }
+      if (
+        page.url.split('/')[3] === 'ver' ||
+        page.url.split('/')[3] !== 'zerotwo' ||
+        page.url.split('/')[3] !== 'animes'
```
Comment (lolamtisch) [inline src/pages/AnimeFenix/main.ts:62]: Not necessary as no overview

**Review by lolamtisch — CHANGES_REQUESTED** [review]

File: src/pages/OtakuFR/tests.json (hunk 9 above)
Comment (Keiser04) [reply src/pages/OtakuFR/tests.json:1]: srry i download all to remplace

**Pull request closed without merging**